modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -59,21 +46,27 @@ def run_task_instance(task, *targets)
       )
     end
 
+    use_args = task_args.nil? ? {} : task_args
+    # TODO: Is it OK to refer to the Puppet::Pal namespace here?
+    Puppet::Pal::TaskSignature.new(task).runnable_with?(use_args) do |mismatch|
+      raise Puppet::ParseError, mismatch
```

Comment:
Is there a more appropriate error type than the `Puppet::ParseError` we could report when the specified task arguments don't match the task definition?